jetson/nav/gate_search/gateStateMachine.cpp

```diff
@@ -167,10 +170,11 @@ NavState GateStateMachine::executeGateTurn()
     return NavState::GateTurn;
 } // executeGateTurn()
 
+// Drive to determined waypoint
 NavState GateStateMachine::executeGateDrive()
 {
-    if( mPhoebe->roverStatus().target2().distance >= 0 ||
-        ( mPhoebe->roverStatus().target().distance >= 0 && mPhoebe->roverStatus().target().id != lastKnownPost1.id ))
+    if( mPhoebe->roverStatus().rightTarget().distance >= 0 ||
+        ( mPhoebe->roverStatus().leftTarget().distance >= 0 && mPhoebe->roverStatus().leftTarget().id != lastKnownPost1.id ))
```

Comment:
1. `lastKnownPost1` should be something more like `lastKnownRightTarget` (consistency for right/left not 1 vs. 2 and consistency between calling it a target vs a post when we mean the same thing), right? Double check it's right, not left, please.
2. This line is really long. If it's over 100 (max 120), I would say make this two lines.
3. This is pretty tricky since you are checking both distances and IDs. Please add a comment explaining why.

Reply:
You appear to be doing this a lot. Would it make sense to abstract this into a function?

Reply:
The lines with distance and ID checking were previously in there for the old gate drive code (not sure who had done that previously), so I kept it.

Reply:
Got it. If you don't think that it's good to do this in your change, make an issue for it. If you do make a separate issue, make it more general than this. Something like "Clean up duplicated code" and in that have an item that is specifically these lines.

Reply:
Will do. I think it would be better to have a separate issue for it.